src/experimental/riff-raff-yaml-file/deployments/s3.ts

```diff
@@ -0,0 +1,61 @@
+import type { GuStack } from "../../../constructs/core";
+import type { GuS3OriginBucket } from "../../../constructs/s3";
+import type { RiffRaffDeployment } from "../types";
+
+export function uploadStaticFilesDeployment(
+  stages: string[],
+  originBucket: GuS3OriginBucket,
+  { name: cfnDeployName }: RiffRaffDeployment
+): RiffRaffDeployment {
+  const { app, _deploymentProps } = originBucket;
+  const { withoutStageAwareness, baseParameterName, cacheControl, surrogateControl, mimeTypes } = _deploymentProps;
+  const { stack, region } = originBucket.stack as GuStack;
+
+  const cacheControlValue = Object.entries(cacheControl).map(([pattern, duration]) => {
+    return {
+      pattern,
+      value: `public, max-age=${duration.toSeconds()}`,
+    };
+  });
+
+  const surrogateControlValue = Object.entries(surrogateControl).map(([pattern, duration]) => {
+    return {
+      pattern,
+      value: `public, max-age=${duration.toSeconds()}`,
+    };
+  });
+
+  return {
+    name: ["upload-static-files", region, stack, app].join("-"),
+    props: {
+      type: "aws-s3",
+      actions: ["uploadStaticFiles"],
+      dependencies: [cfnDeployName],
+      stacks: new Set([stack]),
+      regions: new Set([region]),
+      app,
+      contentDirectory: app,
+      parameters: {
+        bucketSsmKeyStageParam: stages.reduce(
```

Comment:
For ease, always set `bucketSsmKeyStageParam`, even if the same bucket is across stages, for example:

Stage aware:
```yaml
bucketSsmKeyStageParam:
  CODE: /CODE/test/my-bucket/shared-bucket-origin-bucket
  PROD: /PROD/test/my-bucket/shared-bucket-origin-bucket
```

Shared:
```yaml
bucketSsmKeyStageParam:
  CODE: /test/shared-bucket/shared-bucket-origin-bucket
  PROD: /test/shared-bucket/shared-bucket-origin-bucket
```

Reply:
Sharing wouldn't work as we'd deploy CFN in both stages, and both will want to create the Parameter on `/test/shared-bucket/shared-bucket-origin-bucket`, which would fail!

Reply:
> Sharing wouldn't work as we'd deploy CFN in both stages, and both will want to create the Parameter on `/test/shared-bucket/shared-bucket-origin-bucket`, which would fail!

Given this, it seems like we should remove the `withoutStageAwareness` prop, and the ternary expression here which checks it?

```
[stage]: withoutStageAwareness ? baseParameterName : `/${stage}${baseParameterName}`,
```

Or am I misunderstanding the above comment?